File: Algorithm/Python/961. N-Repeated Element in Size 2N Array.py
# -*- coding: utf-8 -*-
# Created on Mon Dec 24 2018 13:47:29
# Author: WuLC
# EMail: liangchaowu5@gmail.com

class Solution(object):
    def repeatedNTimes(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        appeared = set()
        for num in A:
            if num in appeared:
                return num
            appeared.add(num)
        return -1

File: Algorithm/C++/961. N-Repeated Element in Size 2N Array.cc
/*
 * Created on Mon Dec 24 2018 13:51:49
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

#include <vector>
#include <set>

using std::vector;
using std::set;

class Solution {
 public:
  int repeatedNTimes(vector<int>& A) {
    set<int> appeared;
    for (auto num : A) {
      if (appeared.count(num) > 0)
       return num;
      appeared.insert(num);
    }
    return -1;
  }
};